data-prepper-plugins/s3-source/src/test/java/com/amazon/dataprepper/plugins/source/SqsWorkerTest.java

```diff
@@ -94,6 +127,9 @@ void processSqsMessages_should_return_number_of_messages_processed() {
         assertThat(messagesProcessed, equalTo(1));
         verify(s3Service).addS3Object(any(S3ObjectReference.class));
         verify(sqsClient).deleteMessageBatch(any(DeleteMessageBatchRequest.class));
+        verify(sqsMessagesReceivedCounter).increment(any(Double.class));
+        verify(sqsMessagesDeletedCounter).increment(any(Double.class));
+        verify(sqsMessageDelayTimer).record(any(Duration.class));
```

Comment:
You can use an `ArgumentCaptor` and validate that the time is roughly correctly.

Calculate the expected duration and then make sure it is within a certain range.

e.g.
```
assertThat(actualDelayTimer, greaterThanOrEqualTo(expectedDuration.minusSeconds(1));
assertThat(actualDelayTimer, lessThanOrEqualTo(expectedDuration.plusSeconds(1));
```

Relatedly, it may be good to setup a known distance for the S3 Event in the mocking.